apps/somnia/src/commands/archive/weapon.ts

```diff
@@ -0,0 +1,70 @@
+import { WeaponKey } from '@genshin-optimizer/gi/consts'
+import { allStats } from '@genshin-optimizer/gi/stats'
+import type { Interaction, MessageActionRowComponentBuilder } from 'discord.js'
+import {
+  ActionRowBuilder,
+  EmbedBuilder,
+  StringSelectMenuBuilder,
+  StringSelectMenuOptionBuilder,
+} from 'discord.js'
+import { clean, colors } from '../archive'
+
+export function weaponarchive(
+  interaction: Interaction,
+  id: WeaponKey,
+  name: string,
+  data: any,
+  args: string
+) {
+  const msg: any = {}
+  name = data.name
+  let text = Object.values(data.description).join('\n')
+  //weapon rarity color
+  const rarity = allStats.weapon.data[id].rarity
+  //default r1 5stars and r5 others
+  let refine = '1'
+  //no refinements for 1/2 star weapons
+  if (rarity > 2) {
+    //r1 for 5 star weapons
+    if (rarity > 4) refine = '1'
+    //r5 for 3/4 star weapons
+    else refine = '5'
+    //user input override
+    if (args) refine = args
+    name += ` (R${refine})`
+    //create dropdown menu
+    const options = []
+    for (let i = 1; i <= 5; i++) {
```

Comment:
for future reference, can use `range` util for this `range(1, 5).forEach(i => )`